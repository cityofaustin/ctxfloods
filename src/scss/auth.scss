.AuthPage {
  height: 100%;
  background: midnightblue;
  text-align: center;
  input {
    width: 100%;
    font-size: 2rem;
    margin-top: 1em;
    padding-top: 0.4em;
    padding-bottom: 0.4em;
    padding-left: 0.4em;
  }
  h1, p {
    color: white;
    text-align: center;
    padding-left: 0.5em;
    padding-right: 0.5em;
  }
  form {
    width: 100%;
    padding-left: 3%;
    padding-right: 3%;
  }
  a {
    font-size: 1.5rem;
    color: white;
  }
}

.Auth__form-controls {
  width: fit-content;
  margin: 20px auto;
}

.Auth__submit {
  background-color: deepskyblue;
}

.Auth__error-text {
  color: red;
  text-align: center;
  font-size: 1.5rem;
}

.Auth__waiting {
  display: block !important;
  color: white;
  text-align: center;
}

.button-link {
  color: white;
  font-size: 1.5rem;
  background-color: transparent;
  border: none;
  cursor: pointer;
  text-decoration: underline;
  display: inline;
  margin: 0;
  padding: 0;
}

.button-link:hover,
.button-link:focus {
  text-decoration: none;
}
