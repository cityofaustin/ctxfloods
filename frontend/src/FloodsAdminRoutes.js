import React, { Component } from 'react';
import { Route } from 'react-router-dom';
import Login from './Login';
import Public from './Public';
import PrivateRoute from './PrivateRoute';
import Header from './Header';
import ManageUsers from './ManageUsersPage/ManageUsers';
import CreateUser from './CreateUser';
import AdminCrossingList from './AdminCrossingList';
import NewStatusUpdate from './NewStatusUpdate';
import auth from './services/gqlAuth';

const Protected = () => <h3>Protected</h3>;

class FloodsAdminRoutes extends Component {
  render() {
    return (
        <div>
          <Route path="/" component={Header}/>
          <Route path="/public" component={Public}/>
          <Route path="/crossings" component={AdminCrossingList}/>
          <Route path="/login" component={Login}/>
          <PrivateRoute path="/dashboard/users" component={ManageUsers}
            authenticated={auth.isAuthenticated()}
            authorized={auth.roleAuthorized(['floods_community_admin', 'floods_super_admin'])}
          />
          <PrivateRoute path="/protected" component={Protected} authenticated={auth.isAuthenticated()}/>
          <PrivateRoute path="/updatestatus" component={NewStatusUpdate} authenticated={auth.isAuthenticated()}/>
          <PrivateRoute path="/createuser" component={CreateUser}
            authenticated={auth.isAuthenticated()}
            authorized={auth.roleAuthorized(['floods_community_admin', 'floods_super_admin'])}
          />
        </div>
    );
  }
}

export default FloodsAdminRoutes;
