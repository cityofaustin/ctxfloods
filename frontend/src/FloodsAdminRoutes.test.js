import React from 'react';
import { MemoryRouter } from 'react-router';
import renderer from 'react-test-renderer';
import FloodsAdminRoutes from './FloodsAdminRoutes';
import auth from './services/gqlAuth';
import { ApolloProvider } from 'react-apollo';
import client from './services/apolloClientService';

jest.mock('./services/gqlAuth', () => {
  return {
    isAuthenticated: jest.fn()
  }
});

describe('when the user is logged in', () => {
  beforeAll(() => {
    auth.isAuthenticated.mockReturnValue(true);
  });
  
  it('should render the root page correctly', () => {
    const tree = renderer.create(
      <ApolloProvider client={client}>
        <MemoryRouter initialEntries={[ '/' ]}>
          <FloodsAdminRoutes />
        </MemoryRouter>
      </ApolloProvider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render the public page correctly', () => {
    const tree = renderer.create(
      <ApolloProvider client={client}>
        <MemoryRouter initialEntries={[ '/public' ]}>
          <FloodsAdminRoutes />
        </MemoryRouter>
      </ApolloProvider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render the protected page correctly', () => {
   const tree = renderer.create(
      <ApolloProvider client={client}>
        <MemoryRouter initialEntries={[ '/protected' ]}>
          <FloodsAdminRoutes />
        </MemoryRouter>
      </ApolloProvider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render the create user page correctly', () => {
   const tree = renderer.create(
      <ApolloProvider client={client}>
        <MemoryRouter initialEntries={[ '/createuser' ]}>
          <FloodsAdminRoutes />
        </MemoryRouter>
      </ApolloProvider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  // TODO: Figure out a good way to test loading data from GraphQL

});

describe('when the user is logged out', () => {
  beforeAll(() => {
    auth.isAuthenticated.mockReturnValue(false);
  });

  it('should render the root page correctly', () => {
    const tree = renderer.create(
      <ApolloProvider client={client}>
        <MemoryRouter initialEntries={[ '/' ]}>
          <FloodsAdminRoutes />
        </MemoryRouter>
      </ApolloProvider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render the public page correctly', () => {
    const tree = renderer.create(
      <ApolloProvider client={client}>
        <MemoryRouter initialEntries={[ '/public' ]}>
          <FloodsAdminRoutes />
        </MemoryRouter>
      </ApolloProvider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render the protected page correctly', () => {
    const tree = renderer.create(
      <ApolloProvider client={client}>
        <MemoryRouter initialEntries={[ '/protected' ]}>
          <FloodsAdminRoutes />
        </MemoryRouter>
      </ApolloProvider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render the create user page correctly', () => {
    const tree = renderer.create(
      <ApolloProvider client={client}>
        <MemoryRouter initialEntries={[ '/createuser' ]}>
          <FloodsAdminRoutes />
        </MemoryRouter>
      </ApolloProvider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

});
