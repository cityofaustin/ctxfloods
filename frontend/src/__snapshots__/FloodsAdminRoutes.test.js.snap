exports[`when the user is logged in should render the create user page correctly 1`] = `
<div>
  <p>
    Welcome! 
    <button
      onClick={[Function]}>
      Sign out
    </button>
  </p>
  <ul>
    <li>
      <a
        href="/public"
        onClick={[Function]}>
        Public Page
      </a>
    </li>
    <li>
      <a
        href="/crossings"
        onClick={[Function]}>
        List Crossings (Public)
      </a>
    </li>
    <li>
      <a
        href="/users"
        onClick={[Function]}>
        List Users
      </a>
    </li>
    <li>
      <a
        href="/protected"
        onClick={[Function]}>
        Protected Page
      </a>
    </li>
    <li>
      <a
        href="/createuser"
        onClick={[Function]}>
        Create User (Protected)
      </a>
    </li>
    <li>
      <a
        href="/updatestatus"
        onClick={[Function]}>
        Update Crossing Status (Protected)
      </a>
    </li>
  </ul>
  <div>
    <p>
      Create a User
    </p>
    <form
      onSubmit={[Function]}>
      <input
        onChange={[Function]}
        placeholder="First Name"
        type="text"
        value="" />
      <input
        onChange={[Function]}
        placeholder="Last Name"
        type="text"
        value="" />
      <input
        onChange={[Function]}
        placeholder="Email"
        type="text"
        value="" />
      <input
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value="" />
      <input
        type="submit" />
    </form>
  </div>
</div>
`;

exports[`when the user is logged in should render the protected page correctly 1`] = `
<div>
  <p>
    Welcome! 
    <button
      onClick={[Function]}>
      Sign out
    </button>
  </p>
  <ul>
    <li>
      <a
        href="/public"
        onClick={[Function]}>
        Public Page
      </a>
    </li>
    <li>
      <a
        href="/crossings"
        onClick={[Function]}>
        List Crossings (Public)
      </a>
    </li>
    <li>
      <a
        href="/users"
        onClick={[Function]}>
        List Users
      </a>
    </li>
    <li>
      <a
        href="/protected"
        onClick={[Function]}>
        Protected Page
      </a>
    </li>
    <li>
      <a
        href="/createuser"
        onClick={[Function]}>
        Create User (Protected)
      </a>
    </li>
    <li>
      <a
        href="/updatestatus"
        onClick={[Function]}>
        Update Crossing Status (Protected)
      </a>
    </li>
  </ul>
  <h3>
    Protected
  </h3>
</div>
`;

exports[`when the user is logged in should render the public page correctly 1`] = `
<div>
  <p>
    Welcome! 
    <button
      onClick={[Function]}>
      Sign out
    </button>
  </p>
  <ul>
    <li>
      <a
        href="/public"
        onClick={[Function]}>
        Public Page
      </a>
    </li>
    <li>
      <a
        href="/crossings"
        onClick={[Function]}>
        List Crossings (Public)
      </a>
    </li>
    <li>
      <a
        href="/users"
        onClick={[Function]}>
        List Users
      </a>
    </li>
    <li>
      <a
        href="/protected"
        onClick={[Function]}>
        Protected Page
      </a>
    </li>
    <li>
      <a
        href="/createuser"
        onClick={[Function]}>
        Create User (Protected)
      </a>
    </li>
    <li>
      <a
        href="/updatestatus"
        onClick={[Function]}>
        Update Crossing Status (Protected)
      </a>
    </li>
  </ul>
  <h3>
    Public
  </h3>
</div>
`;

exports[`when the user is logged in should render the root page correctly 1`] = `
<div>
  <p>
    Welcome! 
    <button
      onClick={[Function]}>
      Sign out
    </button>
  </p>
  <ul>
    <li>
      <a
        href="/public"
        onClick={[Function]}>
        Public Page
      </a>
    </li>
    <li>
      <a
        href="/crossings"
        onClick={[Function]}>
        List Crossings (Public)
      </a>
    </li>
    <li>
      <a
        href="/users"
        onClick={[Function]}>
        List Users
      </a>
    </li>
    <li>
      <a
        href="/protected"
        onClick={[Function]}>
        Protected Page
      </a>
    </li>
    <li>
      <a
        href="/createuser"
        onClick={[Function]}>
        Create User (Protected)
      </a>
    </li>
    <li>
      <a
        href="/updatestatus"
        onClick={[Function]}>
        Update Crossing Status (Protected)
      </a>
    </li>
  </ul>
</div>
`;

exports[`when the user is logged out should render the create user page correctly 1`] = `
<div>
  <p>
    You are not logged in.
  </p>
  <ul>
    <li>
      <a
        href="/public"
        onClick={[Function]}>
        Public Page
      </a>
    </li>
    <li>
      <a
        href="/crossings"
        onClick={[Function]}>
        List Crossings (Public)
      </a>
    </li>
    <li>
      <a
        href="/users"
        onClick={[Function]}>
        List Users
      </a>
    </li>
    <li>
      <a
        href="/protected"
        onClick={[Function]}>
        Protected Page
      </a>
    </li>
    <li>
      <a
        href="/createuser"
        onClick={[Function]}>
        Create User (Protected)
      </a>
    </li>
    <li>
      <a
        href="/updatestatus"
        onClick={[Function]}>
        Update Crossing Status (Protected)
      </a>
    </li>
  </ul>
  <div>
    <p>
      You must log in to view the page at 
      /createuser
    </p>
    <form
      onSubmit={[Function]}>
      <input
        onChange={[Function]}
        placeholder="Email"
        type="text"
        value="" />
      <input
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value="" />
      <input
        type="submit" />
    </form>
  </div>
</div>
`;

exports[`when the user is logged out should render the protected page correctly 1`] = `
<div>
  <p>
    You are not logged in.
  </p>
  <ul>
    <li>
      <a
        href="/public"
        onClick={[Function]}>
        Public Page
      </a>
    </li>
    <li>
      <a
        href="/crossings"
        onClick={[Function]}>
        List Crossings (Public)
      </a>
    </li>
    <li>
      <a
        href="/users"
        onClick={[Function]}>
        List Users
      </a>
    </li>
    <li>
      <a
        href="/protected"
        onClick={[Function]}>
        Protected Page
      </a>
    </li>
    <li>
      <a
        href="/createuser"
        onClick={[Function]}>
        Create User (Protected)
      </a>
    </li>
    <li>
      <a
        href="/updatestatus"
        onClick={[Function]}>
        Update Crossing Status (Protected)
      </a>
    </li>
  </ul>
  <div>
    <p>
      You must log in to view the page at 
      /protected
    </p>
    <form
      onSubmit={[Function]}>
      <input
        onChange={[Function]}
        placeholder="Email"
        type="text"
        value="" />
      <input
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value="" />
      <input
        type="submit" />
    </form>
  </div>
</div>
`;

exports[`when the user is logged out should render the public page correctly 1`] = `
<div>
  <p>
    You are not logged in.
  </p>
  <ul>
    <li>
      <a
        href="/public"
        onClick={[Function]}>
        Public Page
      </a>
    </li>
    <li>
      <a
        href="/crossings"
        onClick={[Function]}>
        List Crossings (Public)
      </a>
    </li>
    <li>
      <a
        href="/users"
        onClick={[Function]}>
        List Users
      </a>
    </li>
    <li>
      <a
        href="/protected"
        onClick={[Function]}>
        Protected Page
      </a>
    </li>
    <li>
      <a
        href="/createuser"
        onClick={[Function]}>
        Create User (Protected)
      </a>
    </li>
    <li>
      <a
        href="/updatestatus"
        onClick={[Function]}>
        Update Crossing Status (Protected)
      </a>
    </li>
  </ul>
  <h3>
    Public
  </h3>
</div>
`;

exports[`when the user is logged out should render the root page correctly 1`] = `
<div>
  <p>
    You are not logged in.
  </p>
  <ul>
    <li>
      <a
        href="/public"
        onClick={[Function]}>
        Public Page
      </a>
    </li>
    <li>
      <a
        href="/crossings"
        onClick={[Function]}>
        List Crossings (Public)
      </a>
    </li>
    <li>
      <a
        href="/users"
        onClick={[Function]}>
        List Users
      </a>
    </li>
    <li>
      <a
        href="/protected"
        onClick={[Function]}>
        Protected Page
      </a>
    </li>
    <li>
      <a
        href="/createuser"
        onClick={[Function]}>
        Create User (Protected)
      </a>
    </li>
    <li>
      <a
        href="/updatestatus"
        onClick={[Function]}>
        Update Crossing Status (Protected)
      </a>
    </li>
  </ul>
</div>
`;
